### PR TITLE
Switch from using JSON pointers to JQ filters.

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

**Comment by kshakir** [comment]
Please also review/thumb the 34_hotfix version: #4127

File: filesystems/demo-dos/src/main/scala/cromwell/filesystems/demo/dos/DemoDosResolver.scala [hunk 1]
```diff
@@ -22,8 +22,13 @@ case class DemoDosResolver(config: Config) extends StrictLogging {
   private lazy val objectMapper = new ObjectMapper()
   private lazy val marthaUri = config.getString("demo.dos.martha.url")
   private lazy val marthaRequestJsonTemplate = config.getString("demo.dos.martha.request.json-template")
-  private lazy val marthaResponseJsonPointerString = config.getString("demo.dos.martha.response.json-pointer")
-  private lazy val marthaResponseJsonPointer = JsonPointer.compile(marthaResponseJsonPointerString)
+  private lazy val marthaResponseJqFilterString = config.getString("demo.dos.martha.response.jq-filter")
+  private lazy val marthaResponseJqJsonQuery = JsonQuery.compile(marthaResponseJqFilterString)
```
[the remark this answers is not in the document]
Comment (kshakir) [reply filesystems/demo-dos/src/main/scala/cromwell/filesystems/demo/dos/DemoDosResolver.scala]: They're JQ syntax. As far as I know no one has integrated JQ into Refined yet.

**Comment by kshakir** [comment]
When someone points to a spec/RFC I'll hard code the parsing statically. Until then I don't want to recompile-retest-republish each time someone decides to move a field. I haven't seen one yet.

File: filesystems/demo-dos/src/main/scala/cromwell/filesystems/demo/dos/DemoDosResolver.scala [hunk 2]
```diff
@@ -22,8 +22,13 @@ case class DemoDosResolver(config: Config) extends StrictLogging {
...
+  private lazy val marthaResponseJqScope = {
+    val scope = Scope.newEmptyScope
+    scope.loadFunctions(Thread.currentThread().getContextClassLoader)
```
[the remark this answers is not in the document]
Comment (kshakir) [reply filesystems/demo-dos/src/main/scala/cromwell/filesystems/demo/dos/DemoDosResolver.scala]: That's the way the API says it's supposed to be used. Looks like it's dynamically loading in the default JQ functions?

https://github.com/eiiches/jackson-jq/blob/cc80601685a631c808008eaa9f431b1698b53684/jackson-jq/src/test/java/examples/Usage.java#L25-L30

File: filesystems/demo-dos/src/main/scala/cromwell/filesystems/demo/dos/DemoDosResolver.scala [hunk 3]
```diff
@@ -60,7 +65,11 @@ case class DemoDosResolver(config: Config) extends StrictLogging {
           throwUnexpectedResponse
         }
 
-        val resolvedUrlOption = getResolvedUrl(objectMapper.readTree(content).at(marthaResponseJsonPointer))
+        val contentNode = objectMapper.readTree(content)
+        val filteredNodes = marthaResponseJqJsonQuery(marthaResponseJqScope, contentNode).asScala
+        val resolvedUrlOption = filteredNodes.toStream.map(getResolvedUrl).collectFirst {
+          case Some(url) => url
```
[the remark this answers is not in the document]
Comment (kshakir) [reply filesystems/demo-dos/src/main/scala/cromwell/filesystems/demo/dos/DemoDosResolver.scala]: DOS seems fragile. This is the second time we've had to rebuild after Martha updates.

[the remark this answers is not in the document]
Comment (geoffjentry) [reply filesystems/demo-dos/src/main/scala/cromwell/filesystems/demo/dos/DemoDosResolver.scala]: I'd expect DOS to be changing a lot over the next year

[the remark this answers is not in the document]
Comment (geoffjentry) [reply filesystems/demo-dos/src/main/scala/cromwell/filesystems/demo/dos/DemoDosResolver.scala]: That said, I don't particularly like calling it "demo" but ....

[the remark this answers is not in the document]
Comment (kshakir) [reply filesystems/demo-dos/src/main/scala/cromwell/filesystems/demo/dos/DemoDosResolver.scala]: The filesystem "implementation" is a wrapper around the GCS filesystem, using string-prefix replacements and praying the WDL doesn't actually called `read_lines`. Definitely for demo purposes only. 😉

**Pull request merged**